### ISSUE TITLE
uplifted catatonics still have a catatonic side

### ISSUE THREAD

**Issue by ATH1909** [tgstation/tgstation#49203]
I've recently been doing a chaplain gimmick in which I acquire a bunch of catatonic monkeymen and hit each of them 11 times with a hypertool until one of them rolls the split personality trauma, then curing most of my creation's non-split personality traumas and basking in my abiltiy to play God.

Unfortunately, my creations will still spend half of their time in a catatonic state while their "original" catatonic side is in control. I believe that this is not intended, as indicated in the comments of this segment code:
```
/mob/living/split_personality/Life()
	if(QDELETED(body))
		qdel(src) //in case trauma deletion doesn't already do it

	if((body.stat == DEAD && trauma.owner_backseat == src))
		trauma.switch_personalities()
		qdel(trauma)

	//if one of the two ghosts, the other one stays permanently
	if(!body.client && trauma.initialized)
		trauma.switch_personalities()
		qdel(trauma)

	..()
```
It seems like the catatonic side is somehow passing that client check?

It should also be noted that curing the split personality trauma deletes the "awake" side and leaves the catatonic side in the body, as it is technically the "original" controller of the body.

I've yet to experiment with podcloning my creations, though, and that just might be able to work, since it looks like podcloning works based on your client and I have a theory that the catatonic side (or maybe just the body?) is copying the backseater's client variable somehow.

**Comment by Dennok** [comment]
Seems like this check 
https://github.com/tgstation/tgstation/blob/f91f07bf5a451a2b5d9ced04a8c6042af628f3c1/code/datums/brain_damage/split_personality.dm#L62
block swich when we dont have one of the minds.

**Comment by Dennok** [comment]
https://github.com/tgstation/tgstation/blob/f91f07bf5a451a2b5d9ced04a8c6042af628f3c1/code/datums/brain_damage/split_personality.dm#L54-L59

**Comment by XDTM** [comment]
Adding a check on the on_lose could allow the trauma to keep the backseat personality if the owner is missing.

**Comment by ATH1909** [comment]
> Adding a check on the on_lose could allow the trauma to keep the backseat personality if the owner is missing.

IMO, the trauma should instantly delete itself and put the remaining half in control if one of the halves goes catatonic (and if a half is SSD, it should let the other half have full control until the other half comes back).

**Comment by XDTM** [comment]
Yeah, that approach would definitely make sense.

**Comment by Tharcoonvagh** [comment]
Based

**Comment by cacogen** [comment]
I wonder if this will be the gradual easing into respawning as a different character we need and whether I've tipped off the reactionary idiots by saying this

**Comment by XDTM** [comment]
> 
> 
> I wonder if this will be the gradual easing into respawning as a different (human) character we need and whether I've tipped off the reactionary idiots by saying this

I'm sad that the experimental cloner was a casualty of the cloning removal, it had none of the issues that normal cloning was removed for.

**Comment by Dennok** [comment]
> > I wonder if this will be the gradual easing into respawning as a different (human) character we need and whether I've tipped off the reactionary idiots by saying this
> 
> I'm sad that the experimental cloner was a casualty of the cloning removal, it had none of the issues that normal cloning was removed for.

You can return it.
But it must be done in better and clearer shape. As I think.
In less bad code.

**Comment by dragomagol** [comment]
Fixed by #52118
